src/iterated_integrals.jl

```diff
@@ -5,141 +5,11 @@ by the J.
-  WikA::WikAType
-  WikJ::WikJType
-  μ::randType
-  ζ::randType
-  η::randType
-  kronprod1::WikAType
-  kronprod2::WikAType
-end
-
-function KPWJ_iip(ΔW)
-  WikJ = false .* vec(ΔW) .* vec(ΔW)'
-  WikA = false .* vec(ΔW) .* vec(ΔW)'
-
-  μ = zero(ΔW)
-  ζ = zero(ΔW)
-  η = zero(ΔW)
-
-  kronprod1 = zero(WikA)
-  kronprod2 = zero(WikA)
-
-  KPWJ_iip(WikA,WikJ,μ,ζ,η,kronprod1,kronprod2)
-end
-
-
-function get_iterated_I(dt, dW, dZ, Wik::KPWJ_oop, p=nothing, C=1, γ=1//1)
-  m  = length(dW)
-
-  WikJ = vec(dW) .* vec(dW)'
-  WikA = zero(WikJ)
-
-  p==nothing && (p = Int(floor(C*dt^(1//1-2//1*γ)) + 1))
-
-  # Eq.(20)
-  ρp = (π^2)/6
-  for k in 1:p
-    ρp -= (1/k^2)
-  end
-  ρp = ρp/(2*π^2)
-
-  if dZ==nothing
-    μ = randn(eltype(dW),m)
-  else
-    μ = convert(eltype(dW),sqrt(1/dt))*dZ[1:m]
-  end
-
-  for k in 1:p
-    if dZ==nothing
-      ζ = randn(eltype(dW),m)
-      η = randn(eltype(dW),m)
-    else
-      ζ = convert(eltype(dW),sqrt(1/dt))*(dZ[Int(m*(2*k-1)+1):Int(m*2*k)])
-      η = convert(eltype(dW),sqrt(1/dt))*(dZ[Int(m*2*k+1):Int(m*(2*k+1))])
-    end
-
-    kronprod1 = vec(ζ) .* vec(η+convert(eltype(dW),sqrt(2/dt))*dW)'
-    kronprod2 = vec(η+convert(eltype(dW),sqrt(2/dt))*dW) .* vec(ζ)'
-
-    WikA += (kronprod1 - kronprod2)/k
-  end
-
-  WikA *=  dt/(2*π)
-
-  kronprod1 = vec(μ) .* vec(dW)'
-  kronprod2 = vec(dW).* vec(μ)'
-
-  WikJ = WikJ/2 + WikA + sqrt(dt*ρp)*(kronprod1-kronprod2)  # Ito: (UniformScaling(dt))/2
-  return WikJ
-end
-
-
-function get_iterated_I!(dt, dW, dZ, Wik::KPWJ_iip, p=nothing, C=1, γ=1//1)
-  m  = length(dW)
-  @unpack WikA,WikJ,μ,ζ,η,kronprod1,kronprod2 = Wik
-
-  mul!(WikJ,vec(dW),vec(dW)')
-  fill!(WikA, zero(eltype(WikJ)))
-
-  # Below Eq.(26): truncation
-  p==nothing && (p = Int(floor(C*dt^(1//1-2//1*γ)) + 1))
-  #@show p, dZ
-  # Eq.(20)
-  ρp = (π^2)/6
-  for k in 1:p
-    ρp -= (1/k^2)
-  end
-  ρp = ρp/(2*π^2)
-
-  if dZ==nothing
-    randn!(μ)
-  else
-    μ .= convert(eltype(dW),sqrt(1/dt))*@view(dZ[1:m])
-  end
-
-  # Eq. (21)
-  for k in 1:p
-    if dZ==nothing
-      randn!(ζ)
-      randn!(η)
-    else
-      ζ .= convert(eltype(dW),sqrt(1/dt))*@view(dZ[Int(m*(2*k-1)+1):Int(m*2*k)])
-      η .= convert(eltype(dW),sqrt(1/dt))*@view(dZ[Int(m*2*k+1):Int(m*(2*k+1))])
-    end
-    mul!(kronprod1,vec(ζ),vec(η+convert(eltype(dW),sqrt(2/dt))*dW)')
-    mul!(kronprod2,vec(η+convert(eltype(dW),sqrt(2/dt))*dW),vec(ζ)')
-
-    @.. WikA = WikA + (kronprod1 - kronprod2)/k
-  end
-
-  @.. WikA = WikA*dt/(2*π) # check if factor of 2 is correct
-  mul!(kronprod1,vec(μ),vec(dW)')
-  mul!(kronprod2,vec(dW),vec(μ)')
-  #@show WikA, sqrt(dt*ρp)*(kronprod1-kronprod2) , WikA + sqrt(dt*ρp)*(kronprod1-kronprod2)
-  #@show sqrt(dt*ρp)
-  @.. WikJ = WikJ/2 + WikA + sqrt(dt*ρp)*(kronprod1-kronprod2) # Ito: -UniformScaling(dt))/2
-  return nothing
-end
-
-
-
-# Wiktorssson Approximation
+# Wiktorssson Approximation (for diagonal and commutative noise processes where the Levy area approximation is not required.)
```

Comment:
Yup, agreed. The names go back to the first attempt where the goal was to implement Wiktorsson's algorithm (plus handling the special cases efficiently where the Levy area approximation is not required). I already updated the one that's user-facing
`IIWiktorsson` -> `IILevyArea` (open for better suggestions and suggestions for the other ones).

> Also I'm not really sure why you have different routines for diagonal and commutative noise? They should be the same, but there are some slight differences in your implementation I don't understand.

This is because we represent the diffusion matrix for diagonal noise by a vector, see 
https://diffeq.sciml.ai/dev/tutorials/sde_example/

Reply:
> This is because we represent the diffusion matrix for diagonal noise by a vector, see
> [diffeq.sciml.ai/dev/tutorials/sde_example](https://diffeq.sciml.ai/dev/tutorials/sde_example/)

Wouldn't it be easier to just wrap the return value of g in a `Diagonal`? That way the intention is clearer and the solvers don't need to be special cased.
But ok, that was probably discussed when the interface was created and is set in stone now.